API_CHANGES.md

```diff
@@ -15,6 +15,7 @@
 ## Breaking changes
 
 * Remove the `org.sonarsource.sonarlint.core.serverconnection.ServerPathProvider` class.
+* Remove `severity` and `type` fields from `org.sonarsource.sonarlint.core.rpc.protocol.backend.rules.RuleDefinitionDto` as this class is only used for fetching standalone rule details, which should always have the Clean Code Attribute and Impacts
```

Comment:
We could, but from the `RuleDefinitionDto` we were always supposed to get only the CCT stuff. So, realistically, I think it's easier to just remove it. I could also add them back. WDYT @damien-urruty-sonarsource ?

Reply:
The problem is that it will break compatibility for people who did not migrate to this new stuff

Reply:
Summary of the discussion -> We go with the fields removal and do not keep backward compatibility because all flavors will work on it in the coming days